### PR TITLE
HistogramTimers should be #[must_use]

### LINKED ISSUE #145: LocalHistogramTimer should have #[must_use]
If `LocalHistogramTimer` is not used, it is dropped immediately so that the timer cannot collect useful info. We need to explicitly state that it is `must_use` in case of programmer's mistakes.

### PR DESCRIPTION
**Pull request by breeswish**
`HistogramTimers` should be either kept in a variable (so that it will not be immediately dropped), or manually call its `observe_duration()` interface. In this way, it should be marked as `#[must_use]`.

Resolve #145

### CODE REVIEW

**Comment by breeswish** [comment]
@overvenus PTAL

File: src/histogram.rs [hunk 1]
```diff
@@ -309,6 +309,7 @@ mod coarse {
 /// observe.
 ///
 /// NOTICE: A timer can be observed only once (automatically or manually).
+#[must_use]
```
Comment (overvenus) [inline src/histogram.rs]: I'd like to have some notes.

Comment (Hoverbear) [reply src/histogram.rs]: According to https://doc.rust-lang.org/reference/attributes.html#lint-check-attributes we can do `#[must_use = "message"]` and note this. Can we do this? :grin:

Comment (breeswish) [reply src/histogram.rs]: nice idea!

**Comment by siddontang** [comment]
Ping @breeswish

**Comment by breeswish** [comment]
@overvenus @Hoverbear PTAL again, thanks!

**Pull request merged**